Проверим, что у вас подключена возможность работы с SQL

In [66]:
%load_ext sql
%sql sqlite://
%config SqlMagic.style = '_DEPRECATED_DEFAULT'

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


Создадим таблицу Студентов со следующими полями:
1. Student_id - уникальный идентификатор студента
2. Name - имя студента
3. SecondName - фамилия студента
4. Speciality - специальность студента
5. StudyGroup - группа студента
6. GPA - средний балл

In [67]:
%%sql
DROP TABLE IF EXISTS Students;

CREATE TABLE Students
(
    student_id integer PRIMARY KEY,
    name varchar(50),
    secondName varchar(50),
    speciality varchar(50),
    studyGroup varchar(20),
    GPA decimal(1,1)
);

 * sqlite://
Done.
Done.


[]

Проверим, что таблица создана

In [68]:
%sql SELECT * FROM Students

 * sqlite://
Done.


[]

Заполним таблицу данными

In [69]:
%%sql
INSERT INTO Students VALUES (1, 'Альянов', 'Николай', '01.03.02', '04-25', 3.58);
INSERT INTO Students VALUES (2, 'Киселева', 'Ева', '01.03.02', '01-25', 4.58);
INSERT INTO Students VALUES (3, 'Исаева', 'Анна', NULL, '01-25', 4.75);
INSERT INTO Students VALUES (4, 'Костин', 'Павел', '01.03.02', '01-25', 4.83);
INSERT INTO Students VALUES (5, 'Стоянов', 'Тимофей', '01.03.02', '03-25', 4.18);
INSERT INTO Students VALUES (6, 'Петров', 'Александр', '02.03.03', '06-25', 3.91);
INSERT INTO Students VALUES (7, 'Фатеев', 'Максим', '10.03.01', '05-25', 3.8);
INSERT INTO Students VALUES (8, 'Воронцов', 'Петр',  '10.03.01', '05-25', 3.5);
INSERT INTO Students VALUES (9, 'Агашкина', 'Дарья', '01.03.02', '02-25', 4.18);
INSERT INTO Students VALUES (11, 'Комарова', 'Екатерина', '01.03.02', '02-25', 4.83);
INSERT INTO Students VALUES (12, 'Челыхов', 'Иван', '02.03.03', '06-25', 4.45);
INSERT INTO Students VALUES (13, 'Назаров', 'Александр', '02.03.03', '07-25', 3.73);
INSERT INTO Students VALUES (14, 'Горлова', 'Ксения', '01.03.02', '01-25', 5);
INSERT INTO Students VALUES (17, 'Горин', 'Олег', '01.03.02', '01-25', 3.89);
INSERT INTO Students VALUES (18, 'Остапенко', 'Алексей', '01.03.02', '04-25', 3.81);
INSERT INTO Students VALUES (19, 'Антонникова', 'Валерия', '01.03.02', NULL, 4.89);
INSERT INTO Students VALUES (20, 'Котова', 'Евгения', '10.03.01', '05-25', 4.29);

 * sqlite://
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.
1 rows affected.


[]

Проверим наличие данных

In [70]:
%%sql
SELECT * FROM Students;

 * sqlite://
Done.


[(1, 'Альянов', 'Николай', '01.03.02', '04-25', 3.58),
 (2, 'Киселева', 'Ева', '01.03.02', '01-25', 4.58),
 (3, 'Исаева', 'Анна', None, '01-25', 4.75),
 (4, 'Костин', 'Павел', '01.03.02', '01-25', 4.83),
 (5, 'Стоянов', 'Тимофей', '01.03.02', '03-25', 4.18),
 (6, 'Петров', 'Александр', '02.03.03', '06-25', 3.91),
 (7, 'Фатеев', 'Максим', '10.03.01', '05-25', 3.8),
 (8, 'Воронцов', 'Петр', '10.03.01', '05-25', 3.5),
 (9, 'Агашкина', 'Дарья', '01.03.02', '02-25', 4.18),
 (11, 'Комарова', 'Екатерина', '01.03.02', '02-25', 4.83),
 (12, 'Челыхов', 'Иван', '02.03.03', '06-25', 4.45),
 (13, 'Назаров', 'Александр', '02.03.03', '07-25', 3.73),
 (14, 'Горлова', 'Ксения', '01.03.02', '01-25', 5),
 (17, 'Горин', 'Олег', '01.03.02', '01-25', 3.89),
 (18, 'Остапенко', 'Алексей', '01.03.02', '04-25', 3.81),
 (19, 'Антонникова', 'Валерия', '01.03.02', None, 4.89),
 (20, 'Котова', 'Евгения', '10.03.01', '05-25', 4.29)]

## Задание 1. Напишите SQL-запрос, возвращающий фамилии всех студентов с именем Александр или с баллом больше 4.5

## Задание 2. Вывести только тех студентов, фамилия которых начинается с буквы 'к', а в имени есть буква 'е'.

## Задание 3. Для каждой группы выведите оценку её лучшего студента. Результат отсортировать по убыванию баллов.

## Задание 4. Вывести специальности (+ их средние баллы), количество групп в которых больше одной

## Задание 5. Для всех специальностей вывести средний балл по специальности только для студентов, в чьей фамилии вторая или третья буква 'т', при условии, что количество таких студентов на специальности не менее 2. В обратном случае, специальность выводить не требуется.